jedeschule/spiders/saarland.py

```diff
@@ -7,34 +7,37 @@
 class SaarlandSpider(scrapy.Spider):
     name = "saarland"
     # allowed_domains = ["www.saarland.de/4526.htm"]
-    start_urls = ['http://www.saarland.de/4526.htm']
+    start_urls = ['https://www.saarland.de/schuldatenbank.htm?typ=alle&ort=']
 
     def parse(self, response):
-        for link in response.css(".relatedinfo a ::attr(href)").extract():
-            yield scrapy.Request(response.urljoin(link), callback=self.parse_list)
+        for school in response.css(".accordion h3"):
+            uuid = school.css("::attr(data-id)").extract_first()
+            name = school.css("::text").extract_first()
+            details_url = f"https://www.saarland.de/schuldetails.htm?id={uuid}"
+            request = scrapy.Request(details_url, callback=self.parse_list)
+            request.meta['name'] = name.strip() if name else ""
+            yield request
 
     def parse_list(self, response):
-        for link in response.css(".contentteaserlist_frame a ::attr(href)").extract():
-            yield scrapy.Request(response.urljoin(link), callback=self.parse_list)
-
-        schools = response.css(".boxpadding10")
-        for school in schools:
-            # inspect_response(response, self)
-            data = {}
-            text_content = school.css("::text").extract()
-
-            data['name'] = school.css("h2 ::text").extract_first()
-
-            for index, line in enumerate(text_content):
-                if re.match("^\d+ \w+", line):
-                    data['zip'] = line.strip()
-                    data['street'] = text_content[index-1].strip()
-
-            data['email'] = school.css(".link_email a ::attr(href)").extract_first()
-            data['website'] = school.css(".link_external a ::attr(href)").extract_first()
-
-            data['director'] = school.css("::text").re_first("Schulleiter(?:[ /]?in)?:\s(.*)")
-            data['telephone'] = school.css("::text").re_first("Tel(?:[\.:]+)?\s([\(\) 0-9]+)")
-            data['fax'] = school.css("::text").re_first("Fax(?:[\.:]+)?\s([\(\) 0-9]+)")
-
-            yield data
\ No newline at end of file
+        school = response.css("body")
+        data = {'name': response.meta["name"]}
+
+        # All of the entries except for Homepage follow
+        # the pattern `key:value`. For `Homepage` this is
+        # `key:
+        #  value`
+        # This is why we have some extra code to keep track of this
+        homepage_next = False
+        for line in school.css("::text").extract():
+            parts = line.split(': ')
+            key = parts[0].strip()
+            if len(parts) == 2:
+                value = parts[1].strip()
+                data[key] = value
+            if homepage_next:
+                data["Homepage"] = key
+                homepage_next = False
+            if key == "Homepage":
+                homepage_next = True
```

Comment:
Those conditions are mutually exclusive, right? I wonder whether we could rewrite this with `elif` and the last two cases switched to improve readability:

```python
            if len(parts) == 2:
                value = parts[1].strip()
                data[key] = value
            elif key == "Homepage":
                homepage_next = True
            elif homepage_next:
                data["Homepage"] = key
                homepage_next = False
```

Reply:
That is actually how I thought to build it initially but I think it will not work like that. Our structure is as follows:
```python
['\n    Telefax', '0681 9481822\n        ']
['\n    Homepage', '']
['www.tgbbz2-sb.de']
['\n            \n        ']
['\n    E-Mail', 'info@tgbbz2-saarbruecken.de\n        ']
```
So the line where `Homepage` is the key actually also has `len == 2` which is why I stuck with the regular `if`s here.

Reply:
Since none of us came to a better idea, I consider this as resolved.